### PR TITLE
Feature/upgrade ocaml version

### PR DESCRIPTION
**Pull request by mabiede**
- upgrade ocaml version
- fix package dependencies
- update guardian version
- update lock file
- upgrade yarn

### CODE REVIEW

File: dune [hunk 1]
```diff
@@ -1,6 +1,6 @@
 (env
  (dev
   (flags
-   (:standard -w +A-42-44-45-48-66 -warn-error +A-3))))
+   (:standard -w +A-42-44-45-48-66-70 -warn-error +A-3))))
```
Comment (timohuber) [inline dune:4]: What is warning 70?

File: pool.opam [hunk 2]
```diff
@@ -35,7 +35,7 @@ depends: [
   "ppx_yojson_conv" {>= "0.14.0"}
   "ppx_variants_conv" {= "v0.14.2"}
```
Comment (timohuber) [inline pool.opam:36]: So we could update this package now as well?

File: dune (hunk 1 above)
Comment (mabiede) [reply dune:4]: "missing interface" -> which is sometimes wanted ;-)

File: pool.opam (hunk 2 above)
Comment (mabiede) [reply pool.opam]: thanks, done

**Review by timohuber — APPROVED** [review]

**Pull request merged**